### PR TITLE
fix(generator): `dependencies` ignores custom family provider name (#3687)

### LINKED ISSUE #3687: [riverpod_generator] `dependencies` does not respect `provider_family_name_suffix` option
**Describe the bug**

Compilation error in auto-generated file when using `provider_family_name_suffix` option and specifying FamilyProvider in `dependencies` of `@Riverpod` annotation

**To Reproduce**

- `riverpod: 2.5.1`
- `riverpod_annotation: 2.3.5`
- `riverpod_generator: 2.4.0`

`build.yaml`

```yaml
targets:
  $default:
    builders:
      riverpod_generator:
        options:
          provider_family_name_suffix: "ProviderFamily"
```

`providers.dart`

```dart
import 'package:riverpod_annotation/riverpod_annotation.dart';

part 'providers.g.dart';

@Riverpod(dependencies: [])
int count(CountRef ref, String id) => 0;

@Riverpod(dependencies: [count])
int calc(CalcRef ref, String id) => ref.watch(countProviderFamily(id)) * 2;
```

`providers.g.dart`

```dart
...
/// See also [count].
@ProviderFor(count)
const countProviderFamily = CountFamily();
...
/// See also [calc].
@ProviderFor(calc)
const calcProviderFamily = CalcFamily();

/// See also [calc].
class CalcFamily extends Family<int> {
  ...
  static final Iterable<ProviderOrFamily> _dependencies = <ProviderOrFamily>[
    countProvider // <- expected `countProviderFamily`
  ];

  @override
  Iterable<ProviderOrFamily>? get dependencies => _dependencies;

  static final Iterable<ProviderOrFamily> _allTransitiveDependencies =
      <ProviderOrFamily>{
    countProvider,  // <- expected `countProviderFamily`
    ...?countProvider.allTransitiveDependencies // <- expected `countProviderFamily`
  };
  ...
}
```

**Expected behavior**

The build option is applied to the Provider specified in dependencies.

### PR DESCRIPTION
**Pull request by chrg1001**
## Related Issues

fixes #3687

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).

- [x] I have updated the `CHANGELOG.md` of the relevant packages.
      Changelog files must be edited under the form:

  ```md
  ## Unreleased fix/major/minor

  - Description of your change. (thanks to @yourGithubId)
  ```

- [x] If this contains new features or behavior changes,
      I have updated the documentation to match those changes.

### CODE REVIEW

**Comment by rrousselGit** [comment]
Sorry for the delay. Could you add a test for it? Cf `test/intergration/build_yaml`

**Comment by chrg1001** [comment]
@rrousselGit

added test (and rebased)

File: packages/riverpod_generator/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## Unreleased patch
+
+- fix: `dependencies` ignores build options
```
Comment (rrousselGit) [inline packages/riverpod_generator/CHANGELOG.md:3]:
```suggestion
- `@Riverpod(dependencies: [...])` now respects `build.yaml` options
```

**Comment by rrousselGit** [comment]
LGTM.

For information, this approach still won't work in scenarios where a provider is defined in a separate package, using a different `build.yaml`.

I think that's fairly rare. And I'm looking into placing providers themselves in the annotation instead. So I don't expect this to be solved now, but that's worth mentionning.

**Pull request merged**